src/Security/MemberAuthenticator/LoginHandler.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+namespace SilverStripe\Security\MemberAuthenticator;
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\Session;
+use SilverStripe\Control\RequestHandler;
+use SilverStripe\ORM\ValidationResult;
+use SilverStripe\Security\MemberAuthenticator\Authenticator;
+use SilverStripe\Security\Security;
+use SilverStripe\Security\Member;
+
+/**
+ * Handle login requests from MemberLoginForm
+ */
+class LoginHandler extends RequestHandler
+{
+    protected $authenticator;
+
+    private static $url_handlers = [
+        '' => 'login',
+    ];
+
+    /**
+     * Since the logout and dologin actions may be conditionally removed, it's necessary to ensure these
+     * remain valid actions regardless of the member login state.
+     *
+     * @var array
+     * @config
+     */
+    private static $allowed_actions = [
+        'login',
+        'LoginForm',
+        'dologin',
```

Comment:
the doLogin you mean @dhensby ? It shouldn't be I believe, it's LoginForm that delegates to doLogin, right?